test/tests/api/v2_0/skupack_tests.py

```diff
@@ -0,0 +1,422 @@
+            Api().skus_id_get(identifier=self.__sku_id)
+            rsp = self.__client.last_response
+            assert_not_equal(404, rsp.status, message=rsp.reason)
+            codes.append(rsp)
+        for c in codes:
+            assert_equal(200, c.status, message=c.reason)
+            assert_raises(rest.ApiException, Api().skus_id_get, 'fooey')
+
+    @test(groups=['skus_api2.tests', 'api2_patch_skus'], depends_on_groups=['api2_get_skus_by_id'])
+    def skus_patch(self):
+        """Test PATCH:api/2.0/skus/:identifier"""
+        patch_data = {
+            "name": "Quanta-T55"
+        }
+        Api().skus_patch(self.__sku_id, patch_data)
+        result = self.__client.last_response
+        data = loads(self.__client.last_response.data)
+
+        assert_equal(200, result.status, message=result.reason)
+        assert_equal("Quanta-T55", data['name'])
+
+        try:
+            Api().pollers_patch('does_not_exist', {})
+        except ApiException as e:
+            assert_true(404, e.status)
+
+    @test(groups=['skus_api2.tests', 'api2_delete_skus'], depends_on_groups=['api2_patch_skus'])
+    def delete_skus(self):
+        """Test DELETE:api/2.0/skus/:identifier"""
+
+        Api().skus_id_delete(identifier=self.__sku_id)
+        result = self.__client.last_response
+
+        assert_equal(200, result.status, message=result.reason)
+
+        try:
+            Api().skus_id_delete(identifier='does_not_exist')
+        except ApiException as e:
+            assert_true(404, e.status)
+
+    @test(groups=['skus_api2.tests', 'api2_get_sku_nodes'], depends_on_groups=['api2_delete_skus'])
+    def get_sku_nodes(self):
+        """Test GET /api/2.0/skus/:identifier/nodes"""
+        sku=   {
+            "name": "mytestsku",
+            "rules": [
+                {
+                    "path": "dmi.Base Board Information.Manufacturer",
+                    "contains": " "
+                }
+            ]
+        }
+
+        Api().nodes_get_all()
+        self.__nodes = loads(self.__client.last_response.data)
+        i =0
+        for n in self.__nodes:
+
+            if n.get('type') == 'compute':
+
+                #update the sku rule above (rules[0].name.contains) with a value from the cataloged node
+                node_id = n.get('id')
+                Api().nodes_get_catalog_by_id(identifier=node_id)
+                node_catalog_data =loads(self.__client.last_response.data)
+                #LOG.info('node_manufacturer is  :  {0} '.format((node_catalog_data)))
+                if len(node_catalog_data) > 0:
+                    LOG.info('node_manufacturer is  :  {0} '.format((node_catalog_data)[0].get('data').get("Base Board Information").get("Manufacturer")))
+                    node_manufacturer= (node_catalog_data)[0].get('data').get("Base Board Information").get("Manufacturer").split(" ")[0]
+                    sku['rules'][0]['contains'] = node_manufacturer
+
+                    #POST the new sku
```

Comment:
This is a big code block from this line to line 188, it's not easy to understand the code logic, how about to add some empty line between each unit so that bring convenience to other user.